pint_pandas/pint_array.py

```diff
@@ -872,7 +872,7 @@ def convert_values(param):
 
             if coerce_to_dtype:
                 try:
-                    res = cls.from_1darray_quantity(res)
+                    res = cls.from_1darray_quantity(res, self.dtype.subdtype)
```

Comment:
Passing `self.dtype.subdtype` here doesn't fix the problem that at line 901 `pd.array` can convert a `float64` array into a `Float64` array and line 902 will then take the `subdtype` from that, ignoring the subdtype passed in.

Reply:
could you give a worked example? sounds like you've got a `Quantity( NumpyExtensionArray([1.0, ...])` and I'm not sure how that's happened

Reply:
Here's a worked example:
```
import pandas as pd
import numpy as np
import pint_pandas

xx = pd.Series([1.0, 2.0, np.nan], dtype="pint[km][float64]")
yy = pd.Series([2.0, 3.0, 4.0], dtype="pint[km][float64]")

print(xx.fillna(123.456))

print(xx.fillna(xx + yy))
```
The conversion of `xx+yy` from `pint[km][float64]` to `pint[km][Float64]` makes it unsuitable as a further source to a `fillna` operation.